### PR TITLE
Migrate arktika steps 2

### PR DESCRIPTION
**Pull request by Kyle-Wong**
```
        ctx.Step(`^the operation should (fail|succeed)`, fnTestCtx.theManifestOperationShould)                                                                                               // ACTION: move to kubedog or achieve same results
	ctx.Step(`verify InstanceGroup in "ready" state$`, fnTestCtx.verifyInstanceGroup)                                                                                                    // ACTION: use kubedog syntax
	ctx.Step(`^validate Prometheus Statefulset ([^"]*) in namespace ([^"]*) has volumeClaimTemplates name ([^"]*)$`, fnTestCtx.validatePrometheusVolumeClaimTemplatesName)               // ACTION: move to kubedog
	ctx.Step(`^pods in namespace ([^"]*) with selector ([^"]*) have restart count less than (\d+)$`, fnTestCtx.podsWithSelectorHaveRestartCountLessThan)                                 // ACTION: move to kubedog
	ctx.Step(`^I remember the ([^"]*) time$`, fnTestCtx.iRememberTheTime)                                                                                                                // ACTION: move to kubedog
	ctx.Step(`^I wait (\d+) seconds$`, fnTestCtx.iWaitSeconds)     
```

* The step `the operation should succeed/fail` doesn't migrate cleanly to kubedog. In Kubedog, creating resources will return an error if the resource is not created (for example, if blocked by OPA). In arktika, applying manifests will save the result without failing the overall test and this step would read the last result. I didn't want to change the existing behavior of resource creation, so I added new steps to create resources and expecting error or no error.
* I changed the syntax of `I remember the <x> time` because "remembering" is recalling existing information, but the step is actually just saving the current timestamp and mapping it to a string key.

### CODE REVIEW

File: pkg/kubernetes/kube.go [hunk 1]
```diff
@@ -866,3 +895,116 @@ func (kc *Client) PersistentVolExists(volName, expectedPhase string) error {
...
+
+func (kc *Client) VerifyInstanceGroups() error {
+	igs, err := kc.ListInstanceGroups()
+	if err != nil {
+		return err
+	}
+
+	for _, ig := range igs.Items {
+		if !isInstanceGroupStatus(&ig, "ready") {
+			return errors.New("BDD >> failed while instanceGroup state")
```
Comment (danielhelfand) [inline pkg/kubernetes/kube.go:907]: Should we change this to something else? I realize this is how it was internally, but error is a bit confusing.

File: pkg/kubernetes/kube.go [hunk 2]
```diff
@@ -70,6 +72,22 @@ const (
...
+	// CustomResourceAPIVersion is the version of the instance-manager API/custom resource definition
+	CustomResourceAPIVersion = "v1alpha1"
+	// CustomeResourceDomain is the domain of the instance-manager API/custom resource definition
+	CustomeResourceDomain = "keikoproj.io"
+	// CustomResourceName is the name of the API/custom resource definition
+	CustomResourceName = fmt.Sprintf("%v.%v", CustomResourceGroup, CustomeResourceDomain)
+	// CustomResourceKind is the resource kind (plural) of the instance-manager API/custom resource definition
+	CustomResourceKind    = "instancegroups"
+	InstanceGroupResource = schema.GroupVersionResource{Group: CustomResourceName, Version: CustomResourceAPIVersion, Resource: CustomResourceKind}
+)
```
Comment (danielhelfand) [inline pkg/kubernetes/kube.go:89]: Can we make some of these consts? Maybe InstanceGroupResource/CustomResourceName no, but can everything else be?

Comment (Kyle-Wong) [reply pkg/kubernetes/kube.go]: done

File: pkg/kubernetes/kube.go (hunk 1 above)
Comment (Kyle-Wong) [reply pkg/kubernetes/kube.go]: improved the error messaging

**Pull request merged**